src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const temp = {};
+
+  Object.assign(temp, state);
+
+  const arr = [];
+
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties':
+        const add = action.extraData;
+
+        for (const index in add) {
+          temp[index] = add[index];
+        }
+        arr.push({ ...temp });
+        break;
+
+      case action.type === 'removeProperties':
+        const remove = action.keysToRemove;
+
+        for (const index in remove) {
+          delete temp[remove[index]];
+        }
+        arr.push({ ...temp });
+        break;
+
+      case action.type === 'clear':
+        for (const key in temp) {
+          delete temp[key];
+        }
+        arr.push({ ...temp });
```

Comment:
-If you perform same action in all switch cases - do it just once afterwards.
-name variables meanful
checklist 4,5,6 https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md